github/github.go

```diff
@@ -567,6 +613,17 @@ func (c *Client) RateLimits() (*RateLimits, *Response, error) {
 		return nil, nil, err
 	}
 
+	if response.Resources != nil {
+		c.rateMu.Lock()
+		if response.Resources.Core != nil {
```

Comment:
Just curious... did you add any unit test for this section?  I was interested in seeing how these `.Core` and `.Search` responses were returned.